src/Filters/LinearKalmanFilter.cpp

```diff
@@ -0,0 +1,93 @@
+#include "LinearKalmanFilter.h"
+
+namespace mmfs {
+
+LinearKalmanFilter::LinearKalmanFilter(int measurementSize, int controlSize, int stateSize)
+{
+    if (measurementSize <= 0 || controlSize <= 0 || stateSize <= 0) {
+        throw std::invalid_argument("Matrix sizes must be positive integers.");
+    }
+
+    // Initialize with zeros (new keyword does value-initialization in C++03 5.3.4)
+    X = Matrix(stateSize, 1, new double[stateSize]()); // State vector
+    P = Matrix(stateSize, stateSize, new double[stateSize * stateSize]()); // Error covariance matrix
+    K = Matrix(stateSize, measurementSize, new double[stateSize * measurementSize]()); // Kalman gain
+
+    this->measurementSize = measurementSize;
+    this->controlSize = controlSize;
+    this->stateSize = stateSize;
+}
+
+LinearKalmanFilter::LinearKalmanFilter(Matrix X, Matrix P, Matrix K)
+{
+    this->X = X;
+    this->P = P;
+    this->K = K;
```

Comment:
again I am confused. Kalman gain is computed by the algo at every iteration so when would we ever pass it?